src/utils/hooks/use-animation.ts

```diff
@@ -0,0 +1,53 @@
+import { useRef, useEffect } from 'react'
+import { Animated } from 'react-native'
+
+export const useAnimatedValue = (initialValue: number): Animated.Value => {
+  const ref = useRef(new Animated.Value(initialValue))
+  return ref.current
+}
+
+interface BaseAnimationConfig {
+  initialValue?: number
```

Comment:
For clarity, I'm a bit on the edge whether this should be optional or not.

@alexbchr, is this something you considered?

Reply:
I considered it yet, but most of the time, initialValue should be the same as `toValue`, the only times you will want to set `initialValue` is when you want an animation when the component mounts. Making it optional prevents having constructs like this (if we don't want an animation on mount):
```tsx
const anim = useAnimation({
  initialValue: open ? 1 : 0,
  toValue: open ? 1 : 0,
  // Other props...
})
```

Since most of the time, this will be used to animate prop changes and not to initially animate a component, I would leave `initialValue` as optional.